timesketch/api/v1/resources/llm.py

```diff
@@ -0,0 +1,239 @@
+# Copyright 2025 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Timesketch API endpoint for interacting with LLM features."""
+import logging
+import multiprocessing
+import multiprocessing.managers
+import time
+
+import prometheus_client
+from flask import request, abort, jsonify
+from flask_login import login_required, current_user
+from flask_restful import Resource
+
+from timesketch.api.v1 import resources
+from timesketch.lib import definitions, utils
+from timesketch.lib.definitions import METRICS_NAMESPACE
+from timesketch.lib.llms.providers import manager as llm_manager
+from timesketch.lib.llms.features import manager as feature_manager
+from timesketch.models.sketch import Sketch
+
+logger = logging.getLogger("timesketch.api.llm")
+
+
+class LLMResource(resources.ResourceMixin, Resource):
+    """Resource to interact with LLMs."""
+
+    METRICS = {
+        "llm_requests_total": prometheus_client.Counter(
+            "llm_requests_total",
+            "Total number of LLM requests received",
+            ["sketch_id", "feature"],
+            namespace=METRICS_NAMESPACE,
+        ),
+        "llm_errors_total": prometheus_client.Counter(
+            "llm_errors_total",
+            "Total number of errors during LLM processing",
+            ["sketch_id", "feature", "error_type"],
+            namespace=METRICS_NAMESPACE,
+        ),
+        "llm_duration_seconds": prometheus_client.Summary(
+            "llm_duration_seconds",
+            "Time taken to process an LLM request (in seconds)",
+            ["sketch_id", "feature"],
+            namespace=METRICS_NAMESPACE,
+        ),
+    }
+
+    _LLM_TIMEOUT_WAIT_SECONDS = 30
```

Comment:
I think this should be configurable via the `timesketch.conf` to allow for users with slow AI endpoints (e.g. local ollama) to still experiment with the feature by adjusting this setting.

Reply:
Agree this is a good suggestion, but lets address this in a separate PR (I'll make a note), as it requires making changes to timesketch.conf.